pylhc/data_extract/lsa.py

```diff
@@ -100,42 +140,64 @@ def find_trims_at_time(
         """
         if knobs is None or len(knobs) == 0:
             knobs = self.find_knob_names(accelerator)
-        trims = self.getTrims(beamprocess, knobs, end=acc_time.timestamp())
+        trims = self.getTrims(parameter=knobs, beamprocess=beamprocess, end=acc_time.timestamp())
         trims_not_found = [k for k in knobs if k not in trims.keys()]
         if len(trims_not_found):
             LOG.warning(f"The following knobs were not found in '{beamprocess}': {trims_not_found}")
-        return {trim: trims[trim].data[-1] for trim in trims.keys()}  # return last set value
+        trim_dict = {trim: trims[trim].data[-1] for trim in trims.keys()}  # return last set value
+        for trim, value in trim_dict.items():
+            try:
+                trim_dict[trim] = value.flatten()[-1]  # the very last entry ...
+            except AttributeError:
+                continue  # single value, as expected
+            else:
+                LOG.warning(f"Trim {trim} hat multiple data entries {value}, taking only the last one.")
+        return trim_dict
 
-    def get_beamprocess_info(self, beamprocess: str):
+    def get_beamprocess_info(self, beamprocess: Union[str, object]) -> Dict:
         """
         Get context info of the given beamprocess.
 
         Args:
-            beamprocess (str): Name of the beamprocess.
+            beamprocess (str, object): Name of the beamprocess or Beamprocess object.
 
         Returns:
             Dictionary with context info.
         """
-        bp = self._contextService.findStandAloneBeamProcess(beamprocess)
-        return _beamprocess_to_dict(bp)
+        if isinstance(beamprocess, str):
+            beamprocess = self._contextService.findStandAloneBeamProcess(beamprocess)
+        bp_dict = _beamprocess_to_dict(beamprocess)
+        LOG.debug(str(bp_dict))
```

Comment:
Is this log statement forgotten or wanted?

Reply:
wanted :)